Add docs from ri

univerio opened the pull request:
<!-- (optional) Explain your change, focusing on the details of the solution. This is a great place to call out user-visible changes. -->
As promised, wrote a script to add a whole bunch of docs straight from Ruby (rdoc/ri). The script requires Ruby 2.6 to run (due to the dependency on `RubyVM::AbstractSyntaxTree`) and pulls directly from the ri index from the installed Ruby. 52e03a2995804af39220b696963dcb348a1e08f2 was run against Ruby 2.6.4.

Existing docs are overwritten where available for consistency, but note that the formatting is slightly different. In particular, there are no links, and code blocks are not specifically marked `ruby`. Happy to tweak the output.

Best reviewed commit-by-commit.

### Motivation
<!-- Why make this change? Describe the problem, not the solution. This can also be a link to an issue. -->
Now that we have VSCode integration, would be nice to get more docs on hover.

### Test plan
<!-- If you did not write tests for this change, replace the message below explaining why not. Why we should be confident this change is correct? If you changed the website, please include a screenshot of the proposed changes. -->
This is a comment-only change. However, I didn't go through each of the 30k lines individually as that is quite burdensome. Went through a few files as a smoke test and they seemed okay.

There is the distinct possibility that an intentionally-different comment block will get clobbered. Not sure how to prevent that.

univerio commented on rbi/stdlib/csv.rbi:
```diff
@@ -63,34 +288,72 @@ class CSV < Object
+#     CSV("CSV,data") { |c|
+#       c.read.any? { |a| a.include?("zombies") }
+#     } #=> false
+sig do
```
I moved the `sig` here since it seemed to have been in the wrong place, plus I didn't know where the doc would go if the `sig` and `def` are separated like that.

DarkDimius commented on rbi/core/array.rbi:
```diff
@@ -2,323 +2,237 @@
-# A new array can be created by using the literal constructor `[]` .
-# Arrays can contain different types of objects. For example, the array
-# below contains an
-# [Integer](https://ruby-doc.org/core-2.6.3/Integer.html), a
```
It seems that the script deletes some useful information that was present before, notably links and syntax highlighting for code snippets that are embedded in the docs.

DarkDimius commented on rbi/core/data.rbi:
```diff
@@ -1,6 +1,6 @@
 
-# This is a deprecated class, base class for C extensions using
-# Data\_Make\_Struct or Data\_Wrap\_Struct.
+# This is a deprecated class, base class for C extensions using Data_Make_Struct
```
this will make `Make` italic

univerio commented:
r? @DarkDimius 

I fixed all of the formatting issues I found. A few things to note:

* I opted to make all links link to docs.ruby-lang.org instead of ruby-doc.org. The reason for this is docs.ruby-lang.org is more official; it's literally the output of rdoc when run on the ruby code base. This makes it easier to work with since I can hook into the generation. There are some small, subtle differences between the two but overall they have comparable quality.
* Some .rbi files currently have sorbet-specific comments that can be mistaken for documentation. I've made these comments start with `###` to differentiate.
* I also added a diff mode and inspected the output to make sure I'm not introducing regressions in the rendered output. It's not perfect, but you can inspect this by running `./tools/sync-rdoc.rb --dry-run --diff .`.

DarkDimius commented:
Thank you, I've rebased and updated test files
This is awesome ❤️ !
